nx/lib/nx/defn/composite.ex

```diff
@@ -138,7 +135,7 @@ defmodule Nx.Defn.Composite do
   defp flatten_each(%T{} = tensor, acc, _fun),
     do: [tensor | acc]
 
-  defp flatten_each(number, acc, fun) when is_number(number),
+  defp flatten_each(number, acc, fun) when is_number(number) or is_struct(number, Complex),
     do: [fun.(number) | acc]
 
   defp flatten_each(container, acc, fun),
```

Comment:
@josevalim I couldn't remember exactly what the issue was since I wasn't able to reproduce it through `if` in the `defn_test.exs` file. However, I've explicitly tested and fixed these functions directly.

The `@doc false`  functions below were kept as is because I didn't see any `is_number` nor `%T{}` matching

Reply:
Perfect!